influxdb-logger.groovy

```diff
@@ -960,28 +1197,69 @@ void handleInfluxResponse(hubResponse, closure) {
             app.updateSetting("prefBacklogLimit", (Long) 5000)
         }
 
-        if (state.loggerQueue.size() <= settings.prefBacklogLimit) {
-            if (state.loggerQueue.size() > postCount) {
-                logger("Backlog of ${state.loggerQueue.size()} events", logWarn)
+        loggerQueueSize = state.loggerQueue.size()
+        if (loggerQueueSize <= settings.prefBacklogLimit) {
+            if (loggerQueueSize > postCount) {
+                logger("Backlog of ${loggerQueueSize} events", logWarn)
+            }
+
+            // Update queue size variable if in use
+            if (prefQueueSizeVariable) {
+                setGlobalVar(prefQueueSizeVariable, loggerQueueSize)
             }
 
             // Try again later
             runIn(60, writeQueuedDataToInfluxDb)
             return
         }
 
-        logger("Backlog of ${state.loggerQueue.size()} events exceeds limit of ${settings.prefBacklogLimit}: dropping ${postCount} events", logError)
+        logger("Backlog of ${loggerQueueSize} events exceeds limit of ${settings.prefBacklogLimit}: dropping ${postCount} events", logError)
     }
 
     // Remove the post from the queue
     state.loggerQueue = state.loggerQueue.drop(postCount)
+    loggerQueueSize = state.loggerQueue.size()
+
+    // Update queue size variable if in use
+    if (prefQueueSizeVariable) {
+        setGlobalVar(prefQueueSizeVariable, loggerQueueSize)
+    }
 
     // Go again?
-    if (state.loggerQueue.size()) {
+    if (loggerQueueSize) {
         runIn(1, writeQueuedDataToInfluxDb)
     }
 }
 
+/**
+ *  renameVariable()
+ *
+ *  Handle renaming of hub variables
```

Comment:
This is great! I actually had run into this problem and had to manually migrate data over.

Reply:
renameVariable() is an entry point expected to be implemented by applications using hub variables, and is spontaneously called from outside of the application similar to installed() or updated(). Documentation on using hub variables is available [here](https://docs2.hubitat.com/developer/interfaces/hub-variable-api).

Reply:
Ah got it thanks for the explanation! Otherwise the change looks good.